### ISSUE TITLE
ISSUE 25: Time to add a good sidebar

### ISSUE THREAD

**Issue by Lenoxo** [Lenoxo/mk-dashboard#53]
During these last days, I've been using this sidebar that looks more like a nav:

![Captura de pantalla_20241007_164902](https://github.com/user-attachments/assets/660c30e8-7f44-407e-ba49-1384c76e30d1)

So, for now, focus on:

1. Styling the sidebar to actually look like one.
2. If you have time, try to implement a state to show / hide it when sliding from right to left.

**Comment by Lenoxo** [comment]
For now, I'd like to experiment using this pallete in the nav:

https://colorhunt.co/palette/040d12183d3d5c837493b1a6

**Comment by Lenoxo** [comment]
For now, what I added is:

1. A new component, `MainLayout`, that has a `Header` component and a `Navbar` one.
2. To make the translation animation work in the whole app, I used a div called `appWrapper`, and manage a state of `isMenuMobileOpen`, and changing it using buttons in `Header` and `Navbar`.
3. Finally, I styled it to make it fit to the app style

This PR took more time than expected because I had to research a bit on how sites like Canva and some portfolios made the sidebar

Menu closed:

![imagen](https://github.com/user-attachments/assets/e5f5fe9f-2652-44c8-96eb-6fa1dfe98be4)

Menu open:

![imagen](https://github.com/user-attachments/assets/c5ae6cd3-0c7e-4ba0-a0bc-1991d3be76af)
